src/handlers/pull-reviewer.ts

```diff
@@ -0,0 +1,323 @@
+import { logger } from "../helpers/errors";
+import { formatSpecAndPull } from "../helpers/format-spec-and-pull";
+import { fetchIssue } from "../helpers/issue-fetching";
+import { CodeReviewStatus } from "../types/pull-requests";
+import { fetchRepoLanguageStats, fetchRepoDependencies } from "./ground-truths/fetch-deps";
+import { findGroundTruths } from "./ground-truths/find-ground-truths";
+import { Context } from "../types";
+import { CallbackResult } from "../types/proxy";
+import { parseLooseJson } from "../helpers/loose-json-parsing";
+import { closedByPullRequestsReferences, IssuesClosedByThisPr } from "../helpers/gql-queries";
+
+export class PullReviewer {
+  context: Context;
+  private _oneDay = 24 * 60 * 60 * 1000;
+
+  constructor(context: Context<"pull_request.opened" | "pull_request.ready_for_review">) {
+    this.context = context;
+  }
+
+  /**
+   * Perform initial checks on a pull request to determine if review is needed
+   * @returns CallbackResult indicating the status and reason
+   */
+  async performPullPrecheck(): Promise<CallbackResult> {
+    const { logger, payload } = this.context;
+    const { pull_request } = payload;
+
+    // Check if PR is in draft mode, closed, or if we can perform a review
+    if (pull_request.draft) {
+      return { status: 200, reason: logger.info("PR is in draft mode, no action required").logMessage.raw };
+    } else if (pull_request.state === "closed") {
+      return { status: 200, reason: logger.info("PR is closed, no action required").logMessage.raw };
+    } else if (!(await this.canPerformReview())) {
+      return { status: 200, reason: logger.info("Cannot perform review at this time").logMessage.raw };
+    }
```

Comment:
Maybe I missed it but I didn't see a check that the user that changed PR to `ready_for_review` is the same as PR author

Reply:
I thought that was unnecessary as only maintainers and pull authors can invoke a ready for review and event, this would give maintainers the power to request automated reviews

Reply:
But it will fight with admins because maybe the plugin will convert PR to draft but an admin thinks it's ready so they will mark it as ready to review but the plugin will change it back to draft again